fiber/src/core.ml

```diff
@@ -345,6 +345,21 @@ let rec sequential_iter_seq (seq : _ Seq.t) ~f =
     let* () = f x in
     sequential_iter_seq seq ~f
 
+let map_reduce_seq (type a b) (seq : a Seq.t) ~f
+    (module Monoid : Monoid with type t = b) k =
+  match seq () with
+  | Seq.Nil -> k Monoid.empty
+  | Cons (x, xs) ->
```

Comment:
May be worth treating a sequence of length 1 specially? Not sure.

Reply:
It's not quite worth it here. Because checking the `xs` involves evaluating a thunk, we might do a lot of work to do that check.

Reply:
Right, agreed.